src/picongpu/include/particles/creation/creation.kernel

```diff
@@ -0,0 +1,305 @@
+/**
+ * Copyright 2015-2016 Heiko Burau
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "simulation_defines.hpp"
+#include "particles/Particles.hpp"
+#include "mappings/kernel/AreaMapping.hpp"
+#include "particles/ParticlesInit.kernel"
+#include "mappings/simulation/GridController.hpp"
+#include "simulationControl/MovingWindow.hpp"
+#include "traits/Resolve.hpp"
+#include "math/vector/Int.hpp"
+#include "types.h"
+#include "nvidia/atomic.hpp"
+#include <iostream>
+
+namespace picongpu
+{
+namespace particles
+{
+namespace creation
+{
+
+using namespace PMacc;
+
+/** Functor with main kernel for particle creation
+ *
+ * \tparam ParBoxSource container of the source species
+ * \tparam ParBoxTarget container of the target species
+ * \tparam ParticleCreator instance of the particle creation functor
+ *
+ * - maps the frame dimensions and gathers the particle boxes
+ * - contains / calls the Creator
+ */
+template<class ParBoxSource, class ParBoxTarget, class ParticleCreator>
+struct CreateParticlesKernel
+{
+    ParBoxSource sourceBox;
+    ParBoxTarget targetBox;
+    ParticleCreator particleCreator;
+    uint32_t guardSuperCells;
+
+    CreateParticlesKernel(
+        ParBoxSource sourceBox,
+        ParBoxTarget targetBox,
+        ParticleCreator particleCreator,
+        const uint32_t guardSuperCells) :
+            sourceBox(sourceBox),
+            targetBox(targetBox),
+            particleCreator(particleCreator),
+            guardSuperCells(guardSuperCells)
+    {}
+
+    DINLINE void operator()(const PMacc::math::Int<simDim>& cellIndex)
```

Comment:
Please add a documentation for the parameter and define where is origin of the `cellIIndex` is.